types/apollo-upload-client/index.d.ts

```diff
@@ -1,22 +1,58 @@
+// Type definitions for apollo-upload-client 14.1
 // Project: https://github.com/jaydenseric/apollo-upload-client#readme
-// Definitions by: Edward Sammut Alessi <https://github.com/Slessi>
+// Definitions by: Edward Sammut Alessi <https://github.com/Slessi>, tyankatsu <https://github.com/tyankatsu0105>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.1
+// TypeScript Version: 3.6
 
-import { ApolloLink } from "apollo-link";
-import { HttpOptions } from "apollo-link-http-common";
+// ==============================================================================
+// imports
+// ==============================================================================
 
-export { ReactNativeFile } from "extract-files";
+import { ApolloLink } from '@apollo/client/core';
+import { HttpOptions as OriginHttpOptions } from '@apollo/client/link/http';
+import { isExtractableFile as OriginIsExtractableFile } from 'extract-files';
+
+// ==============================================================================
+// declare
+// ==============================================================================
 
 declare global {
     interface GlobalFetch {
-        fetch: WindowOrWorkerGlobalScope["fetch"];
+        fetch: WindowOrWorkerGlobalScope['fetch'];
     }
 }
 
+// ==============================================================================
+// types
+// ==============================================================================
+
+export type HttpOptions = OriginHttpOptions &
+    Partial<{
+        isExtractableFile: typeof OriginIsExtractableFile;
+        formDataAppendFile: typeof formDataAppendFile;
+    }>;
+
+export interface FormDataFileAppender {
+    formData: FormData;
+    fieldName: string;
+    file: any;
+}
+
+// ==============================================================================
+// export
+// ==============================================================================
+
+export { ReactNativeFile, isExtractableFile } from 'extract-files';
+
+/**
+ * Creates a [terminating Apollo Link](https://www.apollographql.com/docs/link/overview/#terminating-links) capable of file uploads.
+ * @see https://github.com/jaydenseric/apollo-upload-client#function-createuploadlink
+ */
+export function createUploadLink(httpOptions?: HttpOptions): ApolloLink;
+
 /**
- * `createUploadLink` options match `createHttpLink` options
- * @param linkOptions `HttpOptions`
+ * The default implementation for [`createUploadLink`](https://github.com/jaydenseric/apollo-upload-client#function-createuploadlink) `options.formDataAppendFile`
+ * that uses the standard [`FormData.append`](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append) method.
+ * @see https://github.com/jaydenseric/apollo-upload-client#function-formdataappendfile
  */
-export function createUploadLink(linkOptions?: HttpOptions): ApolloLink;
+export function formDataAppendFile(formDataFileAppender?: FormDataFileAppender): void;
```

Comment:
https://github.com/jaydenseric/apollo-upload-client/blob/master/src/public/formDataAppendFile.js

are you sure about this definition?

```js
module.exports = function formDataAppendFile(formData, fieldName, file) {
  formData.append(fieldName, file, file.name);
};

Reply:
It's also my mistake.
thanks!

Reply:
done